AlgebraInLean/World01/Sheet7.lean

```diff
@@ -0,0 +1,107 @@
+import AlgebraInLean.World01.Sheet6
+open Defs
+
+/- The last group we will cover in this section is the symmetric group. This is
+defined by the different ways of permuting n elements. The group operation is
+composition. For example, S3 has 6 elements, which permute (1, 2, 3) into
+(1, 2, 3), (1, 3, 2), (2, 1, 3), (2, 3, 1), (3, 1, 2), and (3, 2, 1). Elements
+in Sn can also be thought of as bijective, or invertable, functions. We can
+write an element in Sn as a function from Fin n to Fin n, along with a proof
+that that functoin is bijective.-/
+def Sn (n : ℕ) : Type := {f : Fin n → Fin n // Function.Bijective f}
```

Comment:
If you use the `Equiv` type instead of `Function.Bijective`, you can get rid of a lot of noncomputable definitions. I also think you'd be able to rely more on Mathlib, which might be more instructive.